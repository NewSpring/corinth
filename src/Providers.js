import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
