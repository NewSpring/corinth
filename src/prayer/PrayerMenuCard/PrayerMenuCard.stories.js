import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { PaddedView } from '@apollosproject/ui-kit';

import PrayerMenuCard from '.';

storiesOf('prayer/PrayerMenuCard', module).add('Example', () => (
  <PaddedView>
    <PrayerMenuCard
      title={'A Card Title'}
      image={'https://picsum.photos/600/400/?random'}
      link={'https://github.com'}
      overlayColor={['#6BAC43', '#6BAC43']}
    />
  </PaddedView>
));
