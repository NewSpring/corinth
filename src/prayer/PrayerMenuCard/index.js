import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { styled } from '@apollosproject/ui-kit';
import PrayerMenuTileImage from '../PrayerMenuTileImage';

const Tile = styled(({ theme, selected }) => ({
  marginHorizontal: theme.sizing.baseUnit * -0.75,
  transform: selected
    ? [{ scaleX: 1 }, { scaleY: 1.1 }]
    : [{ scaleX: 0.95 }, { scaleY: 1 }],
}))(View);

const PrayerMenuCard = ({ image, selected, title }) => (
  <Tile selected={selected}>
    <PrayerMenuTileImage image={image} text={title} />
  </Tile>
);

PrayerMenuCard.propTypes = {
  image: PropTypes.string,
  selected: PropTypes.bool,
  title: PropTypes.string,
};

export default PrayerMenuCard;
