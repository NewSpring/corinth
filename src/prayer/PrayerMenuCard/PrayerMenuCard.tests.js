import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import PrayerMenuCard from '.';

const cardData = {
  title: 'A Card Title',
  image: 'https://picsum.photos/600/400/?random',
  overlayColor: ['#6BAC43', '#6BAC43'],
};

describe('the PrayerMenuCard component', () => {
  it('renders a prayer menu card', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerMenuCard
          title={cardData.title}
          image={cardData.image}
          overlayColor={cardData.overlayColor}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
