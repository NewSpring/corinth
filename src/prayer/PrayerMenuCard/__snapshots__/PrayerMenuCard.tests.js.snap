// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PrayerMenuCard component renders a prayer menu card 1`] = `
<View
  style={
    Object {
      "marginHorizontal": -12,
      "transform": Array [
        Object {
          "scaleX": 0.95,
        },
        Object {
          "scaleY": 1,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "height": 150,
        "marginHorizontal": 16,
        "marginVertical": 12,
        "shadowColor": "rgba(8, 0, 20, 0.25)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
        "width": 150,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 16,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <View
          animate={null}
          customAnimate={null}
          forceRatio={1}
          onReady={false}
          style={
            Object {
              "aspectRatio": 1,
              "backgroundColor": "#dddddd",
              "resizeMode": "cover",
              "width": "100%",
            }
          }
        >
          <Image
            maxAspectRatio={1}
            minAspectRatio={1}
            onLoad={[Function]}
            source={
              Array [
                Object {
                  "cache": "force-cache",
                  "uri": "https://picsum.photos/600/400/?random",
                },
              ]
            }
            style={
              Object {
                "backgroundColor": "#dddddd",
                "opacity": 0,
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          />
        </View>
        <BVLinearGradient
          colors={
            Array [
              3865816131,
              3865816131,
            ]
          }
          endPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          overlayColor="#6bac43"
          overlayType="high"
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "paddingBottom": 24,
            "paddingHorizontal": 16,
            "paddingVertical": 16,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "bottom": 16,
              "left": 16,
              "position": "absolute",
              "right": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            A Card Title
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
